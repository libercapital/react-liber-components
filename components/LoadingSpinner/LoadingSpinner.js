"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

require("core-js/modules/es6.symbol");

require("core-js/modules/es6.array.index-of");

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.object.to-string");

require("core-js/modules/es6.object.keys");

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("./styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var LoadingSpinner = function LoadingSpinner(_ref) {
  var color = _ref.color,
      size = _ref.size,
      thickness = _ref.thickness,
      props = _objectWithoutProperties(_ref, ["color", "size", "thickness"]);

  return _react["default"].createElement(_styles.Container, props, _react["default"].createElement(_styles.Spinner, {
    color: color,
    size: size,
    thickness: thickness
  }, _react["default"].createElement("span", null)));
};

LoadingSpinner.propTypes = {
  color: _propTypes["default"].string,
  size: _propTypes["default"].string,
  thickness: _propTypes["default"].string
};
LoadingSpinner.defaultProps = {
  color: null,
  size: null,
  thickness: null
};
var _default = LoadingSpinner;
exports["default"] = _default;