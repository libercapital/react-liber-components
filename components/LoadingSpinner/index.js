"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _LoadingSpinner["default"];
  }
});

var _LoadingSpinner = _interopRequireDefault(require("./LoadingSpinner"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }