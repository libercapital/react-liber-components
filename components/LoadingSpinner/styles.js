"use strict";

require("core-js/modules/es6.object.define-property");

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.object.to-string");

require("core-js/modules/es6.string.iterator");

require("core-js/modules/es6.weak-map");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Spinner = exports.Container = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _Themes = require("../Themes");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var Container = _styledComponents["default"].span.withConfig({
  displayName: "styles__Container",
  componentId: "sc-3p78uc-0"
})(["position:relative;height:fit-content;width:fit-content;"]);

exports.Container = Container;
var spin = (0, _styledComponents.keyframes)(["0%{transform:rotate(0deg);}100%{transform:rotate(360deg);}"]);

var Spinner = _styledComponents["default"].div.withConfig({
  displayName: "styles__Spinner",
  componentId: "sc-3p78uc-1"
})(["animation:", " 3s linear infinite;&,& span,& span::before{display:block;height:", ";width:", ";}& span,& span::before,& span::after{position:absolute;top:0;left:0;bottom:0;right:0;margin:auto;}& span{clip:", ";animation:", " 1.5s cubic-bezier(0.77,0,0.175,1) infinite;}& span::before{content:'';border-top:", " solid ", ";border-radius:50%;opacity:0.5;animation:", " 1.5s cubic-bezier(0.77,0,0.175,1) infinite;}& span::after{content:'';border:", " solid ", ";border-radius:50%;}"], spin, function (props) {
  return props.size || '32px';
}, function (props) {
  return props.size || '32px';
}, function (props) {
  return "rect(calc(".concat(props.size || '32px', "/2), ").concat(props.size || '32px', ", ").concat(props.size || '32px', ", 0)");
}, spin, function (props) {
  return props.thickness || '3px';
}, function (props) {
  return (0, _Themes.handleThemeFromObject)(props.theme, 'loadingSpinner.color', '#4e4e4e', props.color);
}, spin, function (props) {
  return props.thickness || '3px';
}, function (props) {
  return (0, _Themes.handleThemeFromObject)(props.theme, 'loadingSpinner.color', '#4e4e4e', props.color);
});

exports.Spinner = Spinner;