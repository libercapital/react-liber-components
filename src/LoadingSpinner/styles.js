import styled, { keyframes } from 'styled-components';
import { handleThemeFromObject } from '../Themes';

export const Container = styled.span`
  position: relative;
  height: fit-content;
  width: fit-content;
`;

const spin = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;

export const Spinner = styled.div`
  animation: ${spin} 3s linear infinite;
  &,
  & span,
  & span::before {
    display: block;
    height: ${props => props.size || '32px'};
    width: ${props => props.size || '32px'};
  }
  & span,
  & span::before,
  & span::after {
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    margin: auto;
  }
  & span {
    clip: ${props =>
      `rect(calc(${props.size || '32px'}/2), ${props.size || '32px'}, ${props.size || '32px'}, 0)`};
    animation: ${spin} 1.5s cubic-bezier(0.77, 0, 0.175, 1) infinite;
  }
  & span::before {
    content: '';
    border-top: ${props => props.thickness || '3px'} solid
      ${props => handleThemeFromObject(props.theme, 'loadingSpinner.color', '#4e4e4e', props.color)};
    border-radius: 50%;
    opacity: 0.5;
    animation: ${spin} 1.5s cubic-bezier(0.77, 0, 0.175, 1) infinite;
  }

  & span::after {
    content: '';
    border: ${props => props.thickness || '3px'} solid
      ${props => handleThemeFromObject(props.theme, 'loadingSpinner.color', '#4e4e4e', props.color)};
    border-radius: 50%;
  }
`;
