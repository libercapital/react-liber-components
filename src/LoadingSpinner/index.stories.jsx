import React from 'react';
import LoadingSpinner from './LoadingSpinner';

export default {
  title: 'Components 1.0|Loading/Spinner',

  parameters: {
    jest: ['LoadingSpinner'],
  },
};

export const defaultStory = () => <LoadingSpinner />;

defaultStory.story = {
  name: 'default',
};
