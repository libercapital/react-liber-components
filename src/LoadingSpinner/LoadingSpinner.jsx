import React from 'react';
import PropTypes from 'prop-types';
import { Container, Spinner } from './styles';

const LoadingSpinner = ({ color, size, thickness, ...props }) => (
  <Container {...props}>
    <Spinner color={color} size={size} thickness={thickness}>
      <span />
    </Spinner>
  </Container>
);

LoadingSpinner.propTypes = {
  color: PropTypes.string,
  size: PropTypes.string,
  thickness: PropTypes.string,
};

LoadingSpinner.defaultProps = {
  color: null,
  size: null,
  thickness: null,
};

export default LoadingSpinner;
