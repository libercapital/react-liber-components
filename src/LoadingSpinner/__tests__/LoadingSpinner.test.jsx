import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import LoadingSpinner from '../LoadingSpinner';

describe('Test for LoadingSpinner Component', () => {
  configure({ adapter: new Adapter() });

  it('should render correctly without props', () => {
    const wrapper = shallow(<LoadingSpinner />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render passing props', () => {
    const wrapper = shallow(<LoadingSpinner size="64px" thickness="5px" color="#fafafa" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
