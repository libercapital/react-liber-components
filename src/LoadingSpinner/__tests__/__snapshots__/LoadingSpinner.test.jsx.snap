// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for LoadingSpinner Component should render correctly without props 1`] = `
<styles__Container>
  <styles__Spinner
    color={null}
    size={null}
    thickness={null}
  >
    <span />
  </styles__Spinner>
</styles__Container>
`;

exports[`Test for LoadingSpinner Component should render passing props 1`] = `
<styles__Container>
  <styles__Spinner
    color="#fafafa"
    size="64px"
    thickness="5px"
  >
    <span />
  </styles__Spinner>
</styles__Container>
`;
